### PR TITLE
Enable inspection of request builder properties before preparation

### LINKED ISSUE #60: Provide getters for RequestBuilder
I'd like to propose methods to be able to read fields from `RequestBuilder` before it is prepared (or sent). These would mirror the methods available in `PreparedRequest`:
```rust
fn method(&self) -> &Method;
fn headers(&self) -> &HeaderMap;
fn body(&self) -> &[u8];
```
~(and while these names don't clash with `RequestBuilder`'s current API, I'd personally prefix them as `get_method()`, `get_headers()`, and `get_body()`)~

The goal of these is to allow users to build their own extension traits over `RequestBuilder`. For example, I'm working on an interface that requires signing each request with a cryptographic key, and adding that signature to the request's headers. The most convenient way would be to do something like:
```rust
trait RequestBuilderExt { .. }

impl RequestBuilderExt for RequestBuilder {
    fn sign_message(self) -> Self {
	// Sign the message using some combination of the 
	// request's method, url, and body
	let signature = ..;
	
	// Add the signature header to the request
	self.header("ACCESS-SIGNATURE", signature);
	
	self
    }
}
```
which would let my users simply call `.sign_message()` on a `RequestBuilder` prior to sending it. 

As it stands right now, I can't read existing fields from a `RequestBuilder`, and I can't mutate a `PreparedRequest`.

Would these new APIs be welcome to attohttpc?

(and, thanks for writing attohttpc, of all http clients out there it has my favorite API)

### PR DESCRIPTION
**Pull request by adamreichold**
Closes #60

### CODE REVIEW

**Comment by sbstp** [comment]
LGTM

File: src/request/builder.rs [hunk 1]
```diff
@@ -422,6 +420,39 @@ impl<B: Body> RequestBuilder<B> {
...
+impl<B> RequestBuilder<B> {
+    /// Inspect the properties of this request
+    pub fn inspect(&self) -> RequestInspector<'_, B> {
+        RequestInspector(self)
+    }
+}
+
+/// Allows to inspect the properties of a request before preparing it.
+#[derive(Debug)]
+pub struct RequestInspector<'a, B>(&'a RequestBuilder<B>);
```
Comment (sbstp) [inline src/request/builder.rs:432]: Should be it called `RequestBuilderInspector`? It's technically what it is, but it feels a bit Java-esque.

Comment (adamreichold) [reply src/request/builder.rs:432]: The type name will probably never be spelled out in user programs, so I have no problem with changing the name. However, I do concur that `RequestBuilderInspector` is the road towards `AbstractBaseClassImplFactory` and would personally keep the simpler name.

Comment (sbstp) [reply src/request/builder.rs:432]: Yeah ok, it's not a big deal.

**Pull request merged**